shared/src/sources/balancer/swap.rs

```diff
@@ -123,26 +112,15 @@ impl WeightedPoolRef<'_> {
         .map(|bfp| out_reserves.downscale(bfp))
         .flatten()
     }
-
-    fn checked_get_amount_out(
-        &self,
-        out_token: H160,
-        (in_amount, in_token): (U256, H160),
-    ) -> Option<U256> {
-        let out_amount = self.unchecked_get_amount_out(out_token, (in_amount, in_token))?;
-        // We double check that resulting amount out can symmetrically provide an amount in.
-        self.unchecked_get_amount_in(in_token, (out_amount, out_token))?;
-        Some(out_amount)
-    }
 }
 
 impl BaselineSolvable for WeightedPoolRef<'_> {
     fn get_amount_out(&self, out_token: H160, (in_amount, in_token): (U256, H160)) -> Option<U256> {
-        self.checked_get_amount_out(out_token, (in_amount, in_token))
+        self.get_amount_out_(out_token, (in_amount, in_token))
```

Comment:
~~nit: Not a huge fan of having the method name the exact same as the trait name, as it makes it harder to see which one is being called without additional context.~~

**edit**: Oops, missed the sneaky `_`.

Reply:
Even with the `_`, it doesn't really highlight the fact that it is "checked". Either way, I guess this is fine. Although I kinda liked the name "checked_*"

Reply:
My issue with the name `checked_` is that its quite ambiguous about what is checked. Additionally, both functions have the same signature and return an `Option`.